work12/index.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+const wrap = document.getElementById("js-wrap")
+const ul = document.getElementById("js-list");
+const button = document.getElementById("js-button");
+// const url = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// const url = "https://myjson.dit.upm.es/api/bins/bu5z";
+const url = "https://myjson.dit.upm.es/api/bins/2hj3";
+
+async function getData() {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      const json = await response.json();
+      return json;
+    } else {
+      throw new Error(response.statusText);
+    }
+  } catch (e) {
+    throw new Error(e);
+  }
+}
+
+async function getListData() {
+  loading();
+  let listData;
+  try {
+    listData = await getData();
+  } catch (e) {
+    wrap.textContent = `エラー内容:${e.message}`;
+  } finally {
+    hideLoading();
+  }
+  if (listData.data.length === 0) {
+    wrap.textContent = "data is empty";
+    return;
+  }
+  createElement(listData);
+}
+
+function hideLoading() {
+  ul.style.backgroundImage = "none";
+  ul.style.height = "auto";
+}
+
+function createElement({ data }) {
+  const fragment = document.createDocumentFragment();
+
+  data.forEach((value) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+
+    a.href = value.a;
+    a.textContent = value.text;
+    img.src = value.img;
+    img.alt = value.alt;
+    img.style.width = "30px";
+    img.style.verticalAlign = "middle"
+
+    fragment.appendChild(li).appendChild(a).prepend(img);
+  });
+
+  ul.appendChild(fragment);
+}
```

Comment:
IMO

This one also works for not only creating list but also rendering so you should change name or separate some function. Mr.Morita taught me `create**` is just return element, don't append something.
I think changing name is easier like `renderElement`.

This is the website Mr.Morita gave me as a reference. It's written in English though, but I would share with you. 
[JavaScript best practices](https://www.w3.org/wiki/JavaScript_best_practices)

Reply:
@yuka830 
As for createElement, it was difficult to separate the processing, so I named it renderListElement.
with this,
renderElement is set to renderButtonElement.

commit: 73bfce8

Reply:
Good!
But if you use `renderListElement` which includes inserting element in HTML here, then I think you can make  not like `renderButtonTag` in the same way without separating `createButtonTag` and `renderButtonElement`.
I think it would be better to unify one or the other :)

Reply:
The process and display of button creation are summarized in the renderButtonElement function.

commit: [cac07ad](https://github.com/yasu-pro/js-handson/pull/12/commits/cac07ad5097ac60ae01e46f6024dcf08cfc08d5b)

Reply:
Nits
I think `button_wrap` returns here, what do you think? Because `buttonTag` is inserted into `button_wrap`.

Question
Is there any reason why you are using a snake case instead of a camel case for `button_wrap`?

Reply:
> Nits
> I think button_wrap returns here, what do you think? Because buttonTag is inserted into button_wrap.
For the above content, it was okay without "return".
Excuse me.

commit:[da3cae2](https://github.com/yasu-pro/js-handson/pull/12/commits/da3cae21330883f3208108dbf6e153ac40e4879e)

> Question
> Is there any reason why you are using a snake case instead of a camel case for button_wrap?
I should have unified with camel case. I'm sorry

commit:[0ef9090](https://github.com/yasu-pro/js-handson/pull/12/commits/0ef9090711f3b8ebec81dc17d90acadfb1034fdd)